cpp/src/gandiva/function_registry_string.cc

```diff
@@ -515,7 +515,13 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
 
       NativeFunction("translate", {}, DataTypeVector{utf8(), utf8(), utf8()}, utf8(),
                      kResultNullIfNull, "translate_utf8_utf8_utf8",
-                     NativeFunction::kNeedsContext | NativeFunction::kCanReturnErrors)};
+                     NativeFunction::kNeedsContext | NativeFunction::kCanReturnErrors),
+
+      NativeFunction("encode", {}, DataTypeVector{utf8(), utf8()}, utf8(),
+                     kResultNullIfNull, "gdv_fn_encode", NativeFunction::kNeedsContext),
+
+      NativeFunction("decode", {}, DataTypeVector{utf8(), utf8()}, utf8(),
+                     kResultNullIfNull, "gdv_fn_decode", NativeFunction::kNeedsContext)};
```

Comment:
Calling these encode and decode is misleading. What these functions do is transcode (from one encoding to another).

Reply:
So basically what encode function does is it converts a string(utf8 format) to a to it's corresponding utf16 encoding.
And decode converts from the corresponding utf16 encoding to utf8 format.

Reply:
That doesn't answer the comment, does it? "decode" and "encode" seem both too generic _and_ misleading for what these functions actually do.

Reply:
Do you know how to convert a utf16 string in char *?